kafka_consumer/datadog_checks/kafka_consumer/kafka_consumer.py

```diff
@@ -48,70 +33,12 @@ def __init__(self, name, init_config, instances):
-        if isinstance(kafka_version, str):
-            kafka_version = tuple(map(int, kafka_version.split(".")))
-
-        tls_context = self.get_tls_context()
-        crlfile = self.instance.get('ssl_crlfile', self.instance.get('tls_crlfile'))
-        if crlfile:
-            tls_context.load_verify_locations(crlfile)
-            tls_context.verify_flags |= ssl.VERIFY_CRL_CHECK_LEAF
-
-        return clazz(
-            bootstrap_servers=kafka_connect_str,
-            client_id='dd-agent',
-            request_timeout_ms=self.init_config.get('kafka_timeout', DEFAULT_KAFKA_TIMEOUT) * 1000,
-            # if `kafka_client_api_version` is not set, then kafka-python automatically probes the cluster for
-            # broker version during the bootstrapping process. Note that this returns the first version found, so in
-            # a mixed-version cluster this will be a non-deterministic result.
-            api_version=kafka_version,
-            # While we check for SASL/SSL params, if not present they will default to the kafka-python values for
-            # plaintext connections
-            security_protocol=self.instance.get('security_protocol', 'PLAINTEXT'),
-            sasl_mechanism=self.instance.get('sasl_mechanism'),
-            sasl_plain_username=self.instance.get('sasl_plain_username'),
-            sasl_plain_password=self.instance.get('sasl_plain_password'),
-            sasl_kerberos_service_name=self.instance.get('sasl_kerberos_service_name', 'kafka'),
-            sasl_kerberos_domain_name=self.instance.get('sasl_kerberos_domain_name'),
-            sasl_oauth_token_provider=(
-                OAuthTokenProvider(**self.instance['sasl_oauth_token_provider'])
-                if 'sasl_oauth_token_provider' in self.instance
-                else None
-            ),
-            ssl_context=tls_context,
-        )
-
-    @property
-    def kafka_client(self):
-        if self._kafka_client is None:
-            # if `kafka_client_api_version` is not set, then kafka-python automatically probes the cluster for
-            # broker version during the bootstrapping process. Note that this returns the first version found, so in
-            # a mixed-version cluster this will be a non-deterministic result.
-            kafka_version = self.instance.get('kafka_client_api_version')
-            if isinstance(kafka_version, str):
-                kafka_version = tuple(map(int, kafka_version.split(".")))
-
-            self._kafka_client = self._create_kafka_admin_client(api_version=kafka_version)
-        return self._kafka_client
+        self.client = make_client(self)
 
     def check(self, _):
         """The main entrypoint of the check."""
         self._consumer_offsets = {}  # Expected format: {(consumer_group, topic, partition): offset}
         self._highwater_offsets = {}  # Expected format: {(topic, partition): offset}
-        self.client = make_client(self)
```

Comment:
Moving the `make_client(self)` up to `__init__()` instead is required since the original tests never called the `check()` function.